### ISSUE TITLE
Do not announce the initial subflow's IP in an add_addr

### ISSUE THREAD

**Issue by cpaasch** [multipath-tcp/mptcp#41]
It may be that we have an MPTCP-connection right after initializing an IP-address.

In this case, the address-worker of the fullmesh path-manager may not yet have been executed (we have the 500ms delay to allow ifconfig to finish its work).

It then may happen that the host announces the address of the initial subflow:

```
20:43:24.149908 IP 192.0.2.1.41958 > 192.168.0.1.8080: Flags [S], seq 0, win 32792, options [mss 1460,sackOK,nop,nop,nop,wscale 7,mptcp capable csum {0xe7d0d831f2c4620b}], length 0
20:43:24.149988 IP 192.168.0.1.8080 > 192.0.2.1.41958: Flags [S.], seq 2029182451, ack 1, win 28800, options [mss 1460,nop,nop,sackOK,nop,wscale 7,mptcp capable csum {0xce667e15c1bd6344}], length 0
20:43:24.351335 IP 192.0.2.1.41958 > 192.168.0.1.8080: Flags [.], ack 1, win 257, options [mptcp capable csum {0xe7d0d831f2c4620b,0xce667e15c1bd6344},mptcp dss ack 153074637], length 0
20:43:24.353113 IP 192.168.0.1.8080 > 192.0.2.1.41958: Flags [P.], seq 1:1001, ack 1, win 225, options [mptcp dss ack 486545147 seq 153074637 subseq 1 len 1000 csum 0x4879], length 1000
20:43:24.433420 IP 192.168.0.1.8080 > 192.0.2.1.41958: Flags [.], ack 1, win 225, options [mptcp add-addr id 1 192.168.0.1,mptcp dss ack 486545147], length 0
```

**Comment by cpaasch** [comment]
Reproduce with:

```
0   socket(..., SOCK_STREAM, IPPROTO_TCP) = 3
+0  setsockopt(3, SOL_SOCKET, SO_REUSEADDR, [1], 4) = 0
+0  bind(3, ..., ...) = 0
+0  listen(3, 1) = 0

+0  < S 0:0(0) win 32792 <mss 1460,sackOK,nop,nop,nop,wscale 7,mp_capable a>
+0  > S. 0:0(0) ack 1 <mss 1460,sackOK,nop,nop,nop,wscale 7, mp_capable b> 
+0.2  < . 1:1(0) ack 1 win 257 <mp_capable a b, dss dack4> //=trunc_r64_hmac(b)>  
+0  accept(3, ..., ...) = 4 

+0  write(4, ..., 1000) = 1000
+0  > P. 1:1001(1000) ack 1 <dss dack4 dsn4 >

+1 < . 1:1(0) ack 1001 win 225 <dss dack4>

// Send some data, to get an ack back with space available for an ADD_ADDR
+.1 < P. 1:301(300) ack 1001 win 225 <dss dack4 dsn4>
// Now, here we do not expect an ADD_ADDR. But the kernel adds one
+0  > . 1001:1001(0) ack 301 <dss dack4>
```

**Comment by cpaasch** [comment]
Problem is that in mptcp_fullmesh.c in address_worker, when we receive the event and parse the sockets, we maybe should check if the "new" address belongs to the initial subflow. When doing this check, we must make sure that this subflow is still operational and the initial address has not been first removed (REMOVE_ADDR has been sent) and then added back again.

**Comment by okias** [comment]
I noticed when you reference bug like multipath-tcp/mptcp#41 it seems github is able show commit from different repos.

good example are two commits referring this issue. Both are included in 3.12.x repository, but only one shows up here.
If you will use this annotation, it allows people see, what other repositories has also fixed here mentioned issues, which seems to me nice, what you think? :)

**Comment by cpaasch** [comment]
Yes, I now try to always add the "Fixes:"-tag with the github-issue and the commit-hash. Should have done it for the Build-Fix too.
